pkg/cloud/libvirt/client/network.go

```diff
@@ -55,22 +55,22 @@ func HasDHCP(net libvirtxml.Network) bool {
 }
 
 // Tries to update first, if that fails, it will add it
-func updateOrAddHost(n *libvirt.Network, ip, mac, name string) error {
-	err := updateHost(n, ip, mac, name)
-	if virErr, ok := err.(libvirt.Error); ok && virErr.Code == libvirt.ERR_OPERATION_INVALID && virErr.Domain == libvirt.FROM_NETWORK {
-		return addHost(n, ip, mac, name)
+func updateOrAddHost(virtConn *libvirt.Libvirt, n *libvirt.Network, ip, mac, name string) error {
+	err := updateHost(virtConn, n, ip, mac, name)
+	if virErr, ok := err.(libvirt.Error); ok && virErr.Code == uint32(libvirt.ErrOperationInvalid) {
+		return addHost(virtConn, n, ip, mac, name)
 	}
 	return err
 }
 
 // Adds a new static host to the network
-func addHost(n *libvirt.Network, ip, mac, name string) error {
+func addHost(virtConn *libvirt.Libvirt, n *libvirt.Network, ip, mac, name string) error {
 	xmlDesc, err := getHostXMLDesc(ip, mac, name)
 	if err != nil {
 		return fmt.Errorf("error getting host xml desc: %v", err)
 	}
 	glog.Infof("Adding host with XML:\n%s", xmlDesc)
-	return n.Update(libvirt.NETWORK_UPDATE_COMMAND_ADD_LAST, libvirt.NETWORK_SECTION_IP_DHCP_HOST, -1, xmlDesc, libvirt.NETWORK_UPDATE_AFFECT_CURRENT)
+	return virtConn.NetworkUpdate(*n, uint32(libvirt.NetworkUpdateCommandAddLast), uint32(libvirt.NetworkSectionIPDhcpHost), -1, xmlDesc, libvirt.NetworkUpdateAffectCurrent)
```

Comment:
Switching to digital-ocean libvirt does not magically fix these NetworkUpdate incompatibilities :( https://github.com/dmacvicar/terraform-provider-libvirt/commit/0d74474808fea1b94e3e8cdd06aac35b10a0b596#commitcomment-66167241

Reply:
I can't find code doing either a version check, or a connection feature check in digitalocean/go-libvirt.